app/controllers/incomeSources/add/ForeignPropertyCheckDetailsController.scala

```diff
@@ -0,0 +1,207 @@
+                                                      implicit val itvcErrorHandler: ItvcErrorHandler,
+                                                      implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler) extends ClientConfirmedController
+  with FeatureSwitching {
+
+  lazy val foreignPropertyAccountingMethodUrl: String = controllers.incomeSources.add.routes.ForeignPropertyAccountingMethodController.show().url
+  lazy val agentForeignPropertyAccountingMethodUrl: String = controllers.incomeSources.add.routes.ForeignPropertyAccountingMethodController.showAgent().url
+  lazy val backUrlIndividual: String = foreignPropertyAccountingMethodUrl
+  lazy val backUrlAgent: String = agentForeignPropertyAccountingMethodUrl
+
+  def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean)
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+    val backUrl: String = if (isAgent) backUrlAgent else backUrlIndividual
+    val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+    val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.ForeignPropertyCheckDetailsController.submitAgent() else {
+      controllers.incomeSources.add.routes.ForeignPropertyCheckDetailsController.submit()
+    }
+
+    if (isDisabled(IncomeSources)) {
+      if (isAgent) Future.successful(Redirect(controllers.routes.HomeController.showAgent)) else Future.successful(Redirect(controllers.routes.HomeController.show()))
+    } else {
+        getDetails(user) map {
+          case Right(viewModel) =>
+            Ok(checkForeignPropertyDetails(
+              viewModel,
+              postAction = postAction,
+              isAgent,
+              backUrl = backUrl
+            ))
+          case Left(ex) =>
+            if (isAgent) {
+              Logger("application").error(
+                s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+              errorHandler.showInternalServerError()
+            } else {
+              Logger("application").error(
+                s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+              errorHandler.showInternalServerError()
+            }
+      } recover {
+        case ex: Exception =>
+          if (isAgent) {
+            Logger("application").error(
+              s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+            errorHandler.showInternalServerError()
+          } else {
+            Logger("application").error(
+              s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+            errorHandler.showInternalServerError()
+          }
+      }
+    }
+  }
+
+  def getDetails(implicit user: MtdItUser[_]): Future[Either[Throwable, CheckForeignPropertyViewModel]] = {
```

Comment:
I was expecting the all definition to be wrapped in the Future, but thats fine, as long as we returning correct type, as this would be replaced